server/src/main/java/keywhiz/service/permissions/AutomationClientDelegatingPermissionCheck.java

```diff
@@ -0,0 +1,76 @@
+package keywhiz.service.permissions;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AutomationClientDelegatingPermissionCheck implements PermissionCheck{
+  private static final Logger logger = LoggerFactory.getLogger(
+      AutomationClientDelegatingPermissionCheck.class);
+
+  private PermissionCheck delegate;
```

Comment:
So for this one I don't think we're going to need a delegate, this can just be a "terminal" permission check. I guess I would say that a permission check either has permission logic of its own (which this one does) or relying on a delegate's logic, but not both. The pass-through and always-allow are in the second category since they don't have any (non-trivial) permission check logic of their own.